### ISSUE TITLE
[bug] Large FPS drop using Destruction tool on Max Area

### ISSUE THREAD

**Issue by MichaelHillcox** [Direwolf20-MC/BuildingGadgets#125]
As mentioned in #123 there is a bug / performance issue with the destruction tool when it's area is set to 16x16x16. 

With a Mid -> Pretty Okay gaming PC (GTX 1070, I5, 16GB Ram with 4GB Ram dedicated to MC) and having the max area active on the tool, you will notice an FPS drop of about 90%. It's something due to how you are highlighting the blocks. I've done some messing around and found that it's this method https://github.com/Direwolf20-MC/BuildingGadgets/blob/master/src/main/java/com/direwolf20/buildinggadgets/tools/ToolRenders.java#L363 that is causing the issue. 

A simple fix might be to switch over to using your `renderBox` method or changing the overlay to large box instead of highlighting each block separately and causing render updates each tick.

**Comment by Direwolf20-MC** [comment]
Yea I definitely need to try fixing that :). For now you can press the undo hot key to hide the render.

**Comment by Direwolf20-MC** [comment]
FYI this is a part of the reason it’s still beta. I need to clean this up.

**Comment by Direwolf20-MC** [comment]
So heres the thing, I can draw it as one big box, but then it also draws over things it doesn't actually touch (Like air, chests, bedrock, etc). Or I can draw each box separately, and it causes lag. 

![image](https://user-images.githubusercontent.com/39863894/48042238-bd094480-e14e-11e8-8161-903ec79c1180.png)

**Comment by MajorTuvok** [comment]
In theory you could try to avoid drawing it around blocks, which are sourrnded by opaque cubes...
Though, evaluating that would decrease Performance when selecting an area (but I guess not by that much, checking 6 blocks around shouldn't be that much of a Problem)

Reasoning for this: Usually in a large Area destruction the Player won't see all the blocks of what he/she is destructing. So the unseen blocks (surrounded by opaque cubes) can safely be clipped away.

I know that this doesn't clip all blocks away, but it is a start and much simpler to implement than full clipping.

**Comment by MichaelHillcox** [comment]
@Direwolf20-MC I think highlighting a large area is a good way to go about it due to the performance hit that doing them separately. Of course, you could always start a new `Thread` to handle the rending and do the computation out of the main loop. 

Maybe a solution to fix the highlighting would be to do a large area at the same time as implementing #120. By listing the blocks to be destroyed you could also list a count of tileEntities that won't be destroyed. This shouldn't be too hard to compute and rendering the items on the screen should be as easy and maintaining a `List<ItemStack>` and clear it each tick unless the starting block is the same as the previous starting block.

**Comment by Direwolf20-MC** [comment]
That’s kind of how I do it now. Problem is if I draw one big box around the area it’s not easy to exclude specific blocks. Since it’s one big box. 

I think I’ll keep it how it is for now. Can y’all test and see if it’s better? It should be. But you’ll still notice a hit for large areas. Use the range hotkey to hide the render.

**Comment by Direwolf20-MC** [comment]
Grab 2.4.4 off curse.

**Comment by Phylogeny** [comment]
I just tested this. My PC is not great (especially the graphics, as they're integrated -- I don't even have a dedicated graphics card) and even when rendering all 4096 boxes, my FPS drops to 30 in a flat world.

I also found that rendering all 6 quads has the same effect on FPS as drawing only 1 quad. The solution I implemented was to prevent rendering for blocks with no sides to render (#129). With that, FPS only drops by a few even when set to max range.
